govcd/catalog.go

```diff
@@ -520,6 +579,38 @@ func createItemForUpload(client *Client, createHREF *url.URL, catalogItemName st
 	return ovfUploadUrl, nil
 }
 
+// Initiates creation of item in catalog and returns vappTeamplate Url for created item.
+func createItemWithLink(client *Client, createHREF *url.URL, catalogItemName, itemDescription, vappTemplateRemoteUrl string) (*url.URL, error) {
+	util.Logger.Printf("[TRACE] createItemWithLink: %s, item name: %s, description: %s, vappTemplateRemoteUrl: %s \n", createHREF, catalogItemName, itemDescription, vappTemplateRemoteUrl)
+	reqBody := bytes.NewBufferString(
```

Comment:
It is very hard to read this template:
* Is it not possible to use a struct for that?
* Or at least a single template string in backticks `` and `Sprintf` to be something like (not ideal, but a bit more readable):

```go
	reqTemplate := `<UploadVAppTemplateParams xmlns="%s" name="%s" sourceHref="%s">
<Description>%s</Description>
</UploadVAppTemplateParams>
`
	body := fmt.Sprintf(reqTemplate, types.XMLNamespaceVCloud, catalogItemName, vappTemplateRemoteUrl, itemDescription)
	reqBody := bytes.NewBufferString(body)
```

Reply:
Good point, made similar to your idea.